app/src/main/java/de/tum/in/tumcampusapp/component/other/settings/SettingsFragment.java

```diff
@@ -101,6 +111,34 @@ public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
         view.setBackgroundColor(Color.WHITE);
     }
 
+    private void initCafeteriaCardSelections() {
+        CafeteriaLocalRepository repository = new CafeteriaLocalRepository(TcaDb.getInstance(getContext()));
+        List<Cafeteria> cafeterias = repository.getAllCafeteriasNow();
+        Collections.sort(cafeterias, Cafeteria::compareTo);
+
+        String[] cafeteriaNames = new String[cafeterias.size() + 1];
+        String[] cafeteriaIds = new String[cafeterias.size() + 1];
+
+        cafeteriaNames[0] = getString(R.string.settings_cafeteria_depending_on_location);
+        cafeteriaIds[0] = CAFETERIA_BY_LOCATION_SETTINGS_ID;
+        Collection<String> defaultValue = new HashSet<>(1);
+        defaultValue.add(CAFETERIA_BY_LOCATION_SETTINGS_ID);
+
+        for (int i = 0; i < cafeterias.size(); i++) {
+            cafeteriaNames[i+1] = cafeterias.get(i).getName();
+            cafeteriaIds[i+1] = Integer.toString(cafeterias.get(i).getId());
+        }
+        MultiSelectListPreference multiSelectPref = new MultiSelectListPreference(getContext());
```

Comment:
Just wondering: Is there any reason to initialize the `MultiSelectListPreference` in code rather than in `settings.xml`?